From 4a82dba89fee39781e50236d46413e460c01c2a3 Mon Sep 17 00:00:00 2001
From: Srinivasan Shanmugam <srinivasan.shanmugam@amd.com>
Date: Fri, 10 Jul 2026 16:55:53 +0530
Subject: [PATCH 04726/15333] drm/amdkfd: Do not fail process debugfs setup on
 debugfs errors
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

debugfs is intended for debugging only, and failures to create debugfs
entries should not affect normal operation.

Remove the check for debugfs_create_dir() in kfd_debugfs_add_process().
If debugfs entries cannot be created, continue without them instead of
reporting an unnecessary error.

Fixes: 325de6bf3328 ("drm/amdkfd: expose pasid of secondary contexts by debugfs")
Reported-by: Dan Carpenter <error27@gmail.com>
Cc: Zhu Lingshan <lingshan.zhu@amd.com>
Cc: Felix Kuehling <felix.kuehling@amd.com>
Signed-off-by: Srinivasan Shanmugam <srinivasan.shanmugam@amd.com>
Reviewed-by: Christian König <christian.koenig@amd.com>
Signed-off-by: Alex Deucher <alexander.deucher@amd.com>
---
 drivers/gpu/drm/amd/amdkfd/kfd_debugfs.c | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/drivers/gpu/drm/amd/amdkfd/kfd_debugfs.c b/drivers/gpu/drm/amd/amdkfd/kfd_debugfs.c
index 464836f2a53f..33e905d59dd9 100644
--- a/drivers/gpu/drm/amd/amdkfd/kfd_debugfs.c
+++ b/drivers/gpu/drm/amd/amdkfd/kfd_debugfs.c
@@ -211,10 +211,6 @@ int kfd_debugfs_add_process(struct kfd_process *p)
 		entry->proc_dentry = debugfs_create_dir(name,
 							primary_entry->proc_dentry);
 	}
-	if (IS_ERR_OR_NULL(entry->proc_dentry)) {
-		ret = entry->proc_dentry ? PTR_ERR(entry->proc_dentry) : -ENOMEM;
-		goto err_free_entry;
-	}
 
 	list_add(&entry->list, &procs);
 	kfd_debugfs_create_pasid_files(p, entry->proc_dentry);
-- 
2.55.0

